tests/datasource/data_connector/test_configured_asset_azure_data_connector.py

```diff
@@ -0,0 +1,1576 @@
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.list_azure_keys",
+    return_value=["alpha-1.csv", "alpha-2.csv", "alpha-3.csv"],
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.BlobServiceClient"
+)
+def test_instantiation_from_a_config_regex_does_not_match_paths(
+    mock_azure_conn, mock_list_keys, mock_emit, empty_data_context_stats_enabled
+):
+    context: DataContext = empty_data_context_stats_enabled
+
+    report_object = context.test_yaml_config(
+        f"""
+        module_name: great_expectations.datasource.data_connector
+        class_name: ConfiguredAssetAzureDataConnector
+        datasource_name: FAKE_DATASOURCE
+        name: TEST_DATA_CONNECTOR
+        default_regex:
+            pattern: beta-(.*)\\.csv
+            group_names:
+                - index
+        container: my_container
+        name_starts_with: ""
+        assets:
+            alpha:
+        azure_options:
+            account_url: my_account_url.blob.core.windows.net
+            credential: my_credential
+    """,
+        return_mode="report_object",
+    )
+
+    assert report_object == {
+        "class_name": "ConfiguredAssetAzureDataConnector",
+        "data_asset_count": 1,
+        "example_data_asset_names": [
+            "alpha",
+        ],
+        "data_assets": {
+            "alpha": {"example_data_references": [], "batch_definition_count": 0},
+        },
+        "example_unmatched_data_references": [
+            "alpha-1.csv",
+            "alpha-2.csv",
+            "alpha-3.csv",
+        ],
+        "unmatched_data_reference_count": 3,
+    }
+    assert mock_emit.call_count == 1
+
+    anonymized_name = mock_emit.call_args_list[0][0][0]["event_payload"][
+        "anonymized_name"
+    ]
+    expected_call_args_list = [
+        mock.call(
+            {
+                "event": "data_context.test_yaml_config",
+                "event_payload": {
+                    "anonymized_name": anonymized_name,
+                    "parent_class": "ConfiguredAssetAzureDataConnector",
+                },
+                "success": True,
+            }
+        ),
+    ]
+    assert mock_emit.call_args_list == expected_call_args_list
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.list_azure_keys",
+    return_value=["alpha-1.csv", "alpha-2.csv", "alpha-3.csv"],
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.BlobServiceClient"
+)
+def test_get_batch_definition_list_from_batch_request_with_illegal_execution_env_name_raises_error(
```

Comment:
If this is about using "something" then the name should be not "illegal execution environment" but "nonexistent_datasource_name" -- to be specific.